tensordict/nn/probabilistic.py

```diff
@@ -453,8 +454,26 @@ def _dist_sample(
             interaction_type = self.default_interaction_type
 
         if interaction_type is InteractionType.MODE:
-            if hasattr(dist, "get_mode"):
-                return dist.get_mode()
+            try:
+                return dist.deterministic_sample
+            except AttributeError:
+                try:
+                    return dist.mean
+                except AttributeError:
+                    raise NotImplementedError(
+                        f"method {type(dist)}.mode is not implemented."
```

Comment:
maybe say you asked deterministic and we fall back to mean and did not find it

Reply:
```suggestion
                        f"method {type(dist)}.deterministic_sample is not implemented."
```

Reply:
@matteobettini it's in the finally below